src/page.rs

```diff
@@ -1,17 +1,16 @@
 use std::fmt::{self, Debug};
+use std::ptr;
 use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering::SeqCst;
-use std::sync::Arc;
-use std::ptr;
 
-use bincode::{serialize, Infinite};
-use serde::{Serialize, Deserialize};
+use bincode::{Infinite, serialize};
+use serde::{Deserialize, Serialize};
 
 use super::*;
 
 const MAX_FRAG_LEN: usize = 7;
 
-pub trait PageMaterializer {
+pub trait PageMaterializer: Send + Sync {
```

Comment:
yeah, it means that anything that implements `PageMaterializer` must also implement `Send` and `Sync`